pkgs/applications/editors/vscode/default.nix

```diff
@@ -79,19 +73,8 @@ in
       '';
 
     postFixup = lib.optionalString (stdenv.hostPlatform.system == "i686-linux" || stdenv.hostPlatform.system == "x86_64-linux") ''
-      patchelf \
-        --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
-        --set-rpath "${rpath}" \
-        $out/lib/vscode/${executableName}
-
-      patchelf \
-        --set-rpath "${rpath}" \
-        $out/lib/vscode/resources/app/node_modules.asar.unpacked/keytar/build/Release/keytar.node
-
-      patchelf \
-        --set-rpath "${rpath}" \
-        "$out/lib/vscode/resources/app/node_modules.asar.unpacked/native-keymap/build/Release/\
-      keymapping.node"
+      wrapProgram $out/lib/vscode/${executableName} \
+        --prefix LD_LIBRARY_PATH : ${atomEnv.libPath}
```

Comment:
Are those libraries loaded via `dlopen`?

Reply:
If so you can add those manually: https://github.com/NixOS/nixpkgs/pull/50596/files#diff-cdcfceb5de8d7eaedeec7ca699f6b918R15
This is still better then adding all libraries to LD_LIBRARY_PATH because some other programs that are started by vscode might break if incompatible libraries are preloaded.

Reply:
Actually I think `$out/lib/vscode/${executableName}` is just a renaming of the `electron` executable so it might be better to link the one generated from `electron`?